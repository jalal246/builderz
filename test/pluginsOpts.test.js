import { resolve } from "path";

import perFile from "./utils";

describe.only.each`
  pkgName                   | desc
  ${"plugin-alias"}         | ${"alias"}
  ${"plugin-multi-entries"} | ${"multi-entries"}
  ${"plugin-css-basic"}     | ${"bundling basic css"}
  ${"plugin-json-basic"}    | ${"bundling json"}
`("tests $pkgName for testing: $desc", ({ pkgName }) => {
  it("passes", async () => {
    const pathPure = resolve(__dirname, "fixtures", pkgName);
    const distPath = resolve(pathPure, "dist");

    await perFile(pathPure, distPath);
  });
});
