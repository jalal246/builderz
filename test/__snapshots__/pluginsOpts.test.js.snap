// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests plugin-alias for testing: alias passes 1`] = `1`;

exports[`tests plugin-alias for testing: alias passes 2`] = `
"var DEBUG = true;

console.log('DEBUG should be true: ', DEBUG);
"
`;

exports[`tests plugin-css-basic for testing: bundling basic css passes 1`] = `11`;

exports[`tests plugin-css-basic for testing: bundling basic css passes 2`] = `
"function index () {
  var el = document.createElement('div');
  el.className = 'testing';
  return el;
}

module.exports = index;
"
`;

exports[`tests plugin-css-basic for testing: bundling basic css passes 3`] = `
".testing {
	display: flex;
	color: red;
	background: blue;
}
"
`;

exports[`tests plugin-css-basic for testing: bundling basic css passes 4`] = `
"function index () {
  var el = document.createElement('div');
  el.className = 'testing';
  return el;
}

export default index;
"
`;

exports[`tests plugin-css-basic for testing: bundling basic css passes 5`] = `
"module.exports=function(){var e=document.createElement(\\"div\\");return e.className=\\"testing\\",e};
//# sourceMappingURL=pluginCssBasic.min.cjs.js.map
"
`;

exports[`tests plugin-css-basic for testing: bundling basic css passes 6`] = `
"export default function(){var e=document.createElement(\\"div\\");return e.className=\\"testing\\",e}
//# sourceMappingURL=pluginCssBasic.min.esm.js.map
"
`;

exports[`tests plugin-css-basic for testing: bundling basic css passes 7`] = `
"!function(e,n){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?module.exports=n():\\"function\\"==typeof define&&define.amd?define(n):(e=e||self).pluginCssBasic=n()}(this,function(){return function(){var e=document.createElement(\\"div\\");return e.className=\\"testing\\",e}});
//# sourceMappingURL=pluginCssBasic.min.umd.js.map
"
`;

exports[`tests plugin-css-basic for testing: bundling basic css passes 8`] = `
"(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory() :
	typeof define === 'function' && define.amd ? define(factory) :
	(global = global || self, global.pluginCssBasic = factory());
}(this, (function () {
	function index () {
	  var el = document.createElement('div');
	  el.className = 'testing';
	  return el;
	}

	return index;

})));
//# sourceMappingURL=pluginCssBasic.umd.js.map
"
`;

exports[`tests plugin-multi-entries for testing: multi-entries passes 1`] = `1`;

exports[`tests plugin-multi-entries for testing: multi-entries passes 2`] = `
"console.log('i am a');

console.log('i am b');
"
`;
